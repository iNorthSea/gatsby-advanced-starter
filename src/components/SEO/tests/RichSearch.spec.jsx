import RichSearchTags, {
  getAuthorMetadata,
  generateOrganizationMetadata,
  generateArticleMetadata,
} from "../RichSearch";
import {
  tagListHasEmptyValues,
  sampleData,
  containsEmptyValues,
} from "./TestUtils";

describe("seo module RichSearchTags", () => {
  it("generates correct tags on website pages", () => {
    expect.assertions(1);

    const generatedTags = RichSearchTags({
      seoData: sampleData.seoWebsite,
      userData: sampleData.user,
      postData: null,
      orgData: sampleData.organization,
    });

    expect(generatedTags).toMatchSnapshot();
  });

  it("generates correct tags on article pages", () => {
    expect.assertions(1);

    const generatedTags = RichSearchTags({
      seoData: sampleData.seoArticle,
      userData: sampleData.user,
      postData: sampleData.post,
      orgData: sampleData.organization,
    });

    expect(generatedTags).toMatchSnapshot();
  });

  it("doesn't generate empty tags", () => {
    expect.assertions(2);

    // Test for website pages
    const generatedWebsiteTags = RichSearchTags({
      seoData: sampleData.seoWebsite,
      userData: sampleData.user,
      postData: null,
      orgData: sampleData.organization,
    });

    expect(tagListHasEmptyValues(generatedWebsiteTags)).toBe(false);

    // Test for article pages
    const generatedArticleTags = RichSearchTags({
      seoData: sampleData.seoArticle,
      userData: sampleData.user,
      postData: sampleData.post,
      orgData: sampleData.organization,
    });

    expect(tagListHasEmptyValues(generatedArticleTags)).toBe(false);
  });
});

describe("seo module RichSearchTags's utility getAuthorMetadata", () => {
  it("generates valid JSONLD", () => {
    expect.assertions(2);

    const generatedTags = getAuthorMetadata(sampleData.user);

    expect(containsEmptyValues(generatedTags)).toBe(false);

    expect(() => JSON.stringify(generatedTags)).not.toThrow(Error);
  });
});

describe("seo module RichSearchTags's utility generateOrganizationMetadata", () => {
  it("generates valid JSONLD", () => {
    expect.assertions(2);

    const generatedTags = generateOrganizationMetadata(sampleData.organization);

    expect(containsEmptyValues(generatedTags)).toBe(false);

    expect(() => JSON.stringify(generatedTags)).not.toThrow(Error);
  });
});

describe("seo module RichSearchTags's utility generateArticleMetadata", () => {
  it("generates valid JSONLD", () => {
    expect.assertions(2);

    const generatedTags = generateArticleMetadata(
      sampleData.post,
      sampleData.user,
      sampleData.organization
    );

    expect(containsEmptyValues(generatedTags)).toBe(false);

    expect(() => JSON.stringify(generatedTags)).not.toThrow(Error);
  });
});
