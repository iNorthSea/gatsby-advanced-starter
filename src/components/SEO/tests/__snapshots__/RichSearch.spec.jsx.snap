// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`seo module RichSearchTags generates correct tags on article pages 1`] = `
Array [
  <script
    type="application/ld+json"
  >
    {"@context":"http://schema.org","@type":"BlogPosting","image":"/logos/logo-1024.png","url":"/posts/url","headline":"Post title","name":"Post title","description":"Post description.","dateCreated":"2021-12-16T23:24:00.000Z","datePublished":"2021-12-16T23:24:00.000Z","dateModified":"2021-12-19T23:24:00.000Z","author":{"@type":"Person","givenName":"First","familyName":"Last","email":"AdvancedUser@example.com","address":"User Location"},"creator":{"@type":"Person","givenName":"First","familyName":"Last","email":"AdvancedUser@example.com","address":"User Location"},"publisher":{"@context":"https://schema.org","@type":"Organization","url":"https://gatsby-advanced-starter-demo.netlify.com","name":"Organization Name","description":"Organization description","logo":"/logos/logo-512.png"},"mainEntityOfPage":"True","keywords":["Tag1","Tag2"],"articleSection":"Tech","articleBody":"Post body"}
  </script>,
]
`;

exports[`seo module RichSearchTags generates correct tags on website pages 1`] = `
Array [
  <script
    type="application/ld+json"
  >
    {"@context":"https://schema.org","@type":"Organization","url":"https://gatsby-advanced-starter-demo.netlify.com","name":"Organization Name","description":"Organization description","logo":"/logos/logo-512.png"}
  </script>,
]
`;
